proposals/EFIP-4.md

```diff
@@ -0,0 +1,39 @@
+# [EFIP-4] Processing staking to {Treasury, Node Operators} in eETH
+
+**Author**: Vaibhav Valecha (vaibhav@ether.fi)
+
+**Date**: 2024-05-26
+
+## Summary
+
+The purpose of this EFIP is to optimize reward skimming when paying out the node operator. Every quarter, we pay out 5% of validator rewards to the corresponding node operator. This is a very expensive operation because we need to skim rewards from over 20,000 eigenpods and 20,000 etherfinode contracts, where all the execution and consensus rewards are directed. Furthermore, determining the rewards accrued by the entire protocol is not a straightforward process due to the number of different contracts holding the rewards.
+
+To address this, during the rebase operation, instead of minting shares of 90% of the rewards (with 5% going to the treasury and 5% to the node operator), we mint 100% of the rewards and distribute 10% to the treasury. This would require a feature in the liquidity pool that mint shares to the treasury or EOA based on the rewards accrued to the protocol and node operators which would be called by the etherfiadmin. 
```

Comment:
is this correct statement given that we have mixes of {B-NFT, T-NFT}s, thus, 10% of rewards to LP != (treasury + node operator) rewards?